corehq/apps/dump_reload/interface.py

```diff
@@ -69,3 +73,10 @@ def load_from_file(self, extracted_dump_path, force=False):
             )
 
         return total_object_count, loaded_object_count
+
+
+def _get_gzfile_line_count(file_path):
+    # This has to iterate through the whole file, which takes time (~15 minutes
+    # for an 11G sql.gz file), but it's worth it to predict completion time
```

Comment:
I'd be interested to hear other options for this

Reply:
What you've done here seems to be the cleanest way to do this.

You probably thought of using
```python
    result = subprocess.check_output(f'zcat {filepath} | wc -l')
    return int(result.decode('utf-8'))
```
I'm not sure how much faster that might be. `zcat` and `wc -l` might be faster than using Python. (I tested on a small compressed log file, and `zcat` + `wc -l` was a little bit faster.) Firing up a subprocess would be an initial cost, but would be outweighed by the 11G. But your code is much neater than this.

(And of course although you could get the uncompressed size of the SQL quickly from the gzip metadata, it won't tell you how many objects it contains.)

Reply:
I did try running
```bash
$ time zcat filename | wc -l
```
on the commandline vs this implementation, and it was faster - something like 12 vs 15 minutes.  Since it wasn't an order of magnitude improvement, I thought simplicity wins out, but I don't feel too strongly about it.

Reply:
what if we changed the dump command to include some metadata like total number of docs?

Reply:
That could work - the dump output is a zip of a few different files, so it doesn't sound hard to have a meta file with some json data or something.  I'll look in to that.